EDU-880 Create waveform plugin

ahelmberger opened the pull request:
Closes https://educandu.atlassian.net/browse/EDU-880

adrianaluchian commented on src/plugins/audio-waveform/audio-waveform-editor.js:
```diff
@@ -0,0 +1,136 @@
+        <Divider plain>{t('generateImageDividerText')}</Divider>
+        <FormItem {...formItemLayoutWithoutLabel}>
+          <Button type="primary" onClick={handleCreateWaveform}>
+            {t('Wellenform aus Audio-Datei erstellen')}
```
Wrong text here 😅

ahelmberger replied:
Oh my! I forgot to add the translation. Fixed! Thanks for spotting it.

adrianaluchian commented on src/plugins/audio-waveform/audio-waveform-info.js:
```diff
@@ -0,0 +1,77 @@
+import joi from 'joi';
+import React from 'react';
+import cloneDeep from '../../utils/clone-deep.js';
+import AudioDisplay from './audio-waveform-display.js';
```
`AudioWaveformDisplay`

ahelmberger replied:
Fixed.

adrianaluchian commented on src/plugins/audio-waveform/audio-waveform-generator-dialog.js:
```diff
@@ -0,0 +1,219 @@
+          </div>
+          <div>
+            <span>{t('dialogLocalFilePickerBackgroundColorLabel')}: </span>
+            <ColorPicker color={waveformBackgroundColor} onChange={setWaveformBackgroundColor} width={382} inline />
```
A very particular number, this 382 😄 What does it mean?

ahelmberger replied:
It overrode the default of 320 in all cases :rofl: I changed it so the default is now 382 and the props are not set anywhere. (It's the width of the color swatches dropdown btw).

ahelmberger replied:
> Works very nicely! clap
> 
> Just 2 tiny things I found while testing:
> 
> * the info icon overlaps a bit with the background in the dialog

I removed this tooltip/icon completely now as it was redundant (the same text is shown in the preview area initially)

> * when you select a wrong/non-audio format in the dialog from the CDN, it throws and does not stop the spinner; maybe we should display a message to the user to draw attention to having not picked an audio file and stop the loading spinner

Fixed! There's now proper error handling. Thanks!

adrianaluchian commented:
Works very nicely! 👏 

Just 2 tiny things I found while testing:
- the info icon overlaps a bit with the background in the dialog
<img width="598" alt="Screen Shot 2022-10-19 at 09 47 46" src="https://user-images.githubusercontent.com/8189923/196629391-3f0a88f2-436f-48f4-b743-14e672e6a2fb.png">

- when you select a wrong/non-audio format in the dialog from the CDN, it throws and does not stop the spinner; maybe we should display a message to the user to draw attention to having not picked an audio file and stop the loading spinner

ahelmberger commented:
> Works very nicely! clap
> 
> Just 2 tiny things I found while testing:
> 
> * the info icon overlaps a bit with the background in the dialog

I removed this tooltip/icon completely now as it was redundant (the same text is shown in the preview area initially)

> * when you select a wrong/non-audio format in the dialog from the CDN, it throws and does not stop the spinner; maybe we should display a message to the user to draw attention to having not picked an audio file and stop the loading spinner

Fixed! There's now proper error handling. Thanks!
